server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -511,24 +504,24 @@ class InteractiveSession(
     // If the session crashed because of the error, the session should instead go to dead state.
     // Since these 2 transitions are triggered by different threads, there's a race condition.
     // Make sure we won't transit from dead to error state.
-    val areSameStates = serverSideState.getClass() == newState.getClass()
-    val transitFromInactiveToActive = !serverSideState.isActive && newState.isActive
+    val areSameStates = _state.getClass() == newState.getClass()
+    val transitFromInactiveToActive = !_state.isActive && newState.isActive
     if (!areSameStates && !transitFromInactiveToActive) {
-      debug(s"$this session state change from ${serverSideState} to $newState")
-      serverSideState = newState
+      debug(s"$this session state change from ${_state} to $newState")
+      _state = newState
     }
   }
 
   private def ensureActive(): Unit = synchronized {
-    require(serverSideState.isActive, "Session isn't active.")
+    require(_state.isActive, "Session isn't active.")
     require(client.isDefined, "Session is active but client hasn't been created.")
   }
 
   private def ensureRunning(): Unit = synchronized {
-    serverSideState match {
-      case SessionState.Running() =>
+    _state match {
+      case SessionState.Running() | SessionState.Idle() | SessionState.Busy() => Unit
```

Comment:
Yes, actually in interactive session running is not used, only idle and busy and honored.